docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,258 @@
+
+As mentioned in [gardener/etcd-druid#922](https://github.com/gardener/etcd-druid/issues/922), a clear hibernation signal is needed. Since hibernation is not yet supported in `etcd-druid` and is out of scope for this proposal, we only address the method for maintaining immutability.
+
+#### Proposal
+
+To mitigate the risk of backups becoming mutable during extended hibernation under bucket-level immutability, the authors propose the following approach:
+
+1. **Prerequisite: Take a Final Full Snapshot Before Hibernation**  
+   - Before scaling the etcd cluster down to zero replicas, the etcd controller triggers an [on-demand full snapshot](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/05-etcd-operator-tasks.md#trigger-on-demand-fulldelta-snapshot). This ensures that the latest state of the etcd cluster is captured and securely stored before hibernation commences.
+2. **Periodically Re-Upload the Snapshot**  
+   - Re-uploading the latest full snapshot resets its immutability period in the bucket, thereby keeping the backups protected during hibernation.  
+   - By default, the re-upload schedule is determined by `etcd.spec.backup.fullSnapshotSchedule`. At present, this interval cannot be customized exclusively for re-uploads; future enhancements may introduce a dedicated configuration parameter.  
+   - A new operator task type, `ExtendFullSnapshotImmutabilityTask`, will periodically invoke the `reupload-snapshot` and `garbage-collect` commands.
+
+3. **Enhance `etcd-backup-restore`**  
+   - Introduce new CLI commands:
+     - **`reupload-snapshot`** for re-uploading snapshots.  
+     - **`garbage-collect`** for removing older backups whose immutability period has expired.
+
+By capturing a final full snapshot before hibernation, periodically re-uploading it to preserve immutability, and removing stale backups, etcd backups remain safeguarded against accidental or malicious alterations until the cluster is resumed.
+
+##### Etcd CR API Changes
+
+A new field is introduced in `Etcd.spec.backup.store` to indicate the immutability strategy:
+
+```go
+// StoreSpec defines parameters for storing etcd backups.
+type StoreSpec struct {
+    // ...
+    // Immutability configuration for the backup store.
+    Immutability *ImmutabilitySpec `json:"immutability,omitempty"`
+}
+
+// ImmutabilitySpec defines immutability settings.
+type ImmutabilitySpec struct {
+    // RetentionType indicates the type of immutability approach. For example, "Bucket".
+    RetentionType string `json:"retentionType,omitempty"`
+}
+```
+
+If `immutability` is not specified, `etcd-druid` will assume that the bucket is mutable, and no immutability-related logic applies.
```

Comment:
Just add a line saying that we have defined a new type to allow us future enhancements to the immutability specification.

Reply:
Done